src/encoded/homepage.py

```diff
@@ -0,0 +1,166 @@
+from pyramid.view import view_config
+from snovault.util import debug_log
+from dcicutils.misc_utils import ignored
+from snovault.search.search import (
+    search
+)
+from snovault.search.search_utils import make_search_subreq
+from urllib.parse import urlencode
+import concurrent.futures
+
+
+import structlog
+
+
+log = structlog.getLogger(__name__)
+
+
+def includeme(config):
+    config.add_route('home', '/home')
+    config.scan(__name__)
+
+
+def make_concurrent_search_requests(search_helpers):
+    """ Execute multiple search functions concurrently using a thread pool (since this is I/O bound). """
+    results = [-1] * len(search_helpers)  # watch out for -1 counts as indicative of an error
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = []
+        for i, (func, kwargs) in enumerate(search_helpers):
+            future = executor.submit(func, **kwargs)
+            futures.append((i, future))
+        for i, future in futures:  # block in order so ordering is preserved
+            try:
+                result = future.result()
+                results[i] = result
+            except Exception as e:
+                print(f"Exception occurred in function {search_helpers[i][0].__name__}: {e}")
+    return results
+
+
+def extract_desired_facet_from_search(facets, desired_facet_name):
+    """ Grabs a single facet from a search response facets block """
+    for d in facets:
+        if d['field'] == desired_facet_name:
+            return d
+    return None
+
+
+def generate_colo829_cell_line_file_count(context, request):
```

Comment:
Consider abstracting out the file count and assay count to separate functions to be reused for other datasets more easily.